scripts/lint_prowjobs/main.go

```diff
@@ -0,0 +1,290 @@
+		} else if strings.Contains(presubmitConfig.JobBase.Name, "release-tooling") && jobMakeTarget != jc.ReleaseToolingMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf("Invalid make target, please use the \"%s\" target", jc.ReleaseToolingMakeTarget)
+		} else if jobMakeTarget != jc.DefaultMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf("Invalid make target, please use the \"%s\" target", jc.DefaultMakeTarget)
+		}
+		return true, "", ""
+	})
+}
+
+func PostsubmitMakeTargetCheck(jc *JobConstants) postsubmitCheck {
+	return postsubmitCheck(func(postsubmitConfig config.Postsubmit, fileContentsString string) (bool, string, string) {
+		jobMakeTargetMatches := regexp.MustCompile(`make (\w+) .*`).FindStringSubmatch(strings.Join(postsubmitConfig.JobBase.Spec.Containers[0].Command, " "))
+		jobMakeTarget := jobMakeTargetMatches[len(jobMakeTargetMatches)-1]
+		makeCommandLineNo := findLineNumber(fileContentsString, "make")
+		if jobMakeTarget != jc.DefaultMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf("Invalid make target, please use the \"%s\" target", jc.DefaultMakeTarget)
+		}
+		return true, "", ""
+	})
+}
+
+func getFilesChanged(gitRoot string, pullBaseSha string, pullPullSha string) ([]string, []string, error) {
+	presubmitFiles := []string{}
+	postsubmitFiles := []string{}
+	gitDiffCommand := "git -C " + gitRoot + " diff --name-only " + pullBaseSha + " " + pullPullSha
+	fmt.Println(gitDiffCommand)
+
+	gitDiffOutput, err := exec.Command("git", strings.Split(gitDiffCommand, " ")[1:]...).Output()
+	filesChanged := strings.Fields(string(gitDiffOutput))
+	for _, file := range filesChanged {
+		if strings.Contains(file, "presubmits") {
+			presubmitFiles = append(presubmitFiles, file)
+		}
+		if strings.Contains(file, "postsubmits") {
+			postsubmitFiles = append(postsubmitFiles, file)
+		}
+	}
+	return presubmitFiles, postsubmitFiles, err
+}
+
+func unmarshalJobFile(filePath string, jobConfig *config.JobConfig) (string, string, string, *config.JobConfig) {
+	jobOrgRepo := strings.Replace(filepath.Dir(filePath), "jobs/", "", 1)
+	jobFileName := filepath.Base(filePath)
+	fileContents, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Fatalf("Error reading contents of %s: %v", filePath, err)
+	}
+	fileContentsString := string(fileContents)
+	if err := yaml.Unmarshal(fileContents, &jobConfig); err != nil {
+		log.Fatalf("Error unmarshaling contents of %s: %v", filePath, err)
+	}
+	return jobOrgRepo, jobFileName, fileContentsString, jobConfig
+}
+
+func displayConfigErrors(fileErrorMap map[string][]string) bool {
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 8, 0, '\t', 0)
+	errorsExist := false
+	for file, errors := range fileErrorMap {
+		if len(errors) > 0 {
+			errorsExist = true
+		}
+		fmt.Println()
+		fmt.Println(file + ":")
+		for _, err := range errors {
+			fmt.Fprintln(w, err)
+		}
```

Comment:
* Since you're using tabwriter, you can get alignment even if the filename length is variable
* nit: `err` is conventionally used for `error` types, consider renaming

```suggestion
		for _, err := range errors {
			fmt.Fprintf(w, "%s\t%s\n", file, err)
		}
```

Reply:
I wanted to print the file name only once, and then the list of errors corresponding to that file below, similar to [this](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/test-infra/19852/pull-test-infra-yamllint/1324290471857491968/build-log.txt)